felix/dataplane/linux/policy_mgr.go

```diff
@@ -144,7 +122,46 @@ func (m *policyManager) OnUpdate(msg interface{}) {
 	}
 }
 
+func (m *policyManager) cleanUpPolicy(id *proto.PolicyID) {
+	if m.rawEgressOnly {
+		m.updateNeededIPSets(id, nil)
+	}
+	inName := rules.PolicyChainName(rules.PolicyInboundPfx, id)
+	outName := rules.PolicyChainName(rules.PolicyOutboundPfx, id)
+	// As above, we need to clean up in all the tables.
+	m.filterTable.RemoveChainByName(inName)
+	m.filterTable.RemoveChainByName(outName)
+	m.mangleTable.RemoveChainByName(inName)
+	m.mangleTable.RemoveChainByName(outName)
+	m.rawTable.RemoveChainByName(inName)
+	m.rawTable.RemoveChainByName(outName)
+}
+
+func (m *policyManager) updateNeededIPSets(id *proto.PolicyID, neededIPSets set.Set[string]) {
+	if neededIPSets != nil {
+		m.neededIPSets[*id] = neededIPSets
+	} else {
+		delete(m.neededIPSets, *id)
+	}
+	m.ipSetFilterDirty = true
+}
+
 func (m *policyManager) CompleteDeferredWork() error {
-	// Nothing to do, we don't defer any work.
+	if !m.rawEgressOnly {
```

Comment:
why do we have the `rawEgressOnly` if it is ever only set to true at the time of creation?

Reply:
That's the one that controls what mode the policy manager is in.  Either it's in BPF mode (filtering IP sets) or its in iptables mode.

We don't want to call the callback if we're in iptables mode.  We always want all IP sets in that case.

Reply:
Ohhh, there are 2 new() got it!